src/audio/rsp_mixer.S

```diff
@@ -1050,25 +1050,25 @@ Mix32Start:
 Mix32Loop:
 	# Apply volume/panning to each channel sample.
 	# left channel:
-	vmulf v_mix_l, v_sample_0, v_xvol_l_0,0;                   
-	vmacf v_mix_l, v_sample_1, v_xvol_l_1,0;           # Store previous loop's output         
-	vmacf v_mix_l, v_sample_2, v_xvol_l_2,0;           ssv v_out_l,0, -2,s4
-	vmacf v_mix_l, v_sample_3, v_xvol_l_3,0;           ssv v_out_r,0, -1,s4
+	vmulf v_mix_l, v_sample_0, v_xvol_l_0,e(v);
+	vmacf v_mix_l, v_sample_1, v_xvol_l_1,e(v);        # Store previous loop's output
+	vmacf v_mix_l, v_sample_2, v_xvol_l_2,e(v);        ssv v_out_l,0, -4,s4
+	vmacf v_mix_l, v_sample_3, v_xvol_l_3,e(v);        ssv v_out_r,0, -2,s4
 	# right channel:                                   # Updated counters
-	vmulf v_mix_r, v_sample_0, v_xvol_r_0,0;           add s0, 32*2
-	vmacf v_mix_r, v_sample_1, v_xvol_r_1,0;           addi t0, -1
-	vmacf v_mix_r, v_sample_2, v_xvol_r_2,0;           addi s4, 4
-	vmacf v_mix_r, v_sample_3, v_xvol_r_3,0;
+	vmulf v_mix_r, v_sample_0, v_xvol_r_0,e(v);        add s0, 32*2
+	vmacf v_mix_r, v_sample_1, v_xvol_r_1,e(v);        addi t0, -1
+	vmacf v_mix_r, v_sample_2, v_xvol_r_2,e(v);        addi s4, 4
+	vmacf v_mix_r, v_sample_3, v_xvol_r_3,e(v);
 
 	# Mix all lanes together into the first lane       # Load next loop's samples
-	vaddc v_out_l, v_mix_l, v_mix_l,3;                 lqv v_sample_0,0, 0,s0
-	vaddc v_out_r, v_mix_r, v_mix_r,3;                 lqv v_sample_1,0, 1,s0
+	vaddc v_out_l, v_mix_l, v_mix_l,e(1q);             lqv v_sample_0,0, 0x00,s0
+	vaddc v_out_r, v_mix_r, v_mix_r,e(1q);             lqv v_sample_1,0, 0x10,s0
 	  # 1 cycle stall here
-	vaddc v_out_l, v_out_l, v_out_l,6;                 lqv v_sample_2,0, 2,s0
-	vaddc v_out_r, v_out_r, v_out_r,6;                 lqv v_sample_3,0, 3,s0
+	vaddc v_out_l, v_out_l, v_out_l,e(2h);             lqv v_sample_2,0, 0x20,s0
+	vaddc v_out_r, v_out_r, v_out_r,e(2h);             lqv v_sample_3,0, 0x30,s0
 	  # 1 cycle stall here
-	vaddc v_out_l, v_out_l, v_out_l,12;                bnez t0, Mix32Loop
-	vaddc v_out_r, v_out_r, v_out_r,12;                
+	vaddc v_out_l, v_out_l, v_out_l,e(4);                bnez t0, Mix32Loop
+	vaddc v_out_r, v_out_r, v_out_r,e(4);
```

Comment:
Now that I see like this, I think this sequence (1q / 2h / 4) is a little arbitrary. I think it can be changed to be more symmetrical. Anyway, I'll look into it as a followup, this comment was just to highlight that this macro makes the code much easier to read.